clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractHeartbeatRequestManager.java

```diff
@@ -313,12 +315,32 @@ private void onFailure(final Throwable exception, final long responseTimeMs) {
             logger.debug(message);
         } else {
             logger.error("{} failed due to fatal error: {}", heartbeatRequestName(), exception.getMessage());
-            handleFatalFailure(exception);
+            if (isHBApiUnsupportedErrorMsg(exception)) {
+                // This is expected to be the case where building the request fails because the node does not support
+                // the API. Propagate custom message.
+                String msg = "The cluster does not support the new consumer group protocol. " +
```

Comment:
Should we use the same error message here? We could perhaps have it in a constant.